tests/integ/test_data_source_api.py

```diff
@@ -229,6 +240,17 @@ def test_dbapi_with_temp_table(session):
     assert df.collect() == all_type_data
 
 
+@pytest.mark.skipif(
+    "config.getoption('local_testing_mode', default=False)",
+    reason="feature not available in local testing",
+)
+def test_dbapi_batch_fetch(session):
+    df = session.read.dbapi(
+        create_connection, SQL_SERVER_TABLE_NAME, max_workers=4, fetch_size=1
+    )
+    assert df.collect() == all_type_data
```

Comment:
can you add more comprehensive test code:
1. fetchmany is called (data_size // fetch_size + 1) times, should be achievable by mock
2. a different combination of  data_size + fetch_size